cryostatio/cryostat
# feat(agent): implement HTTP-based readonly queries to -agents

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Fixes #1416 
Fixes #1309
Fixes #520 

## Description of the change:
This fills in the implementation of one readonly query (MBean metrics) that Cryostat can perform against HTTP-based Cryostat Agent instances. This is simply achieved by wrapping around a Vert.x WebClient HTTP request and deserializing the response into the expected data type conforming to the same interface as what the JMX equivalent already does.

## Motivation for the change:
See linked issues. This PR and others to come later provide a non-JMX alternate communications protocol so that Cryostat can be usefully deployed with no requirement on target applications to expose JMX connections.

## How to manually test:
1. *Run CRYOSTAT_IMAGE=quay.io... sh smoketest.sh...*
2. Check out [the corresponding `-agent` PR](https://github.com/cryostatio/cryostat-agent/pull/78), `mvn install` it, and rebuild `quarkus-test` sample application as usual.
3. Open web-client dashboard, select the `http://` test application, and ensure that MBean Metrics chart card data displays.
